enhancements/cert-manager-proxy/http01_challenge_proxy.md

```diff
@@ -0,0 +1,364 @@
+    // that redirects traffic from the API endpoint on port 80 to ingress routers.
+    // This enables cert-manager to perform HTTP01 ACME challenges for API endpoint certificates.
+    // +optional
+    HTTP01ChallengeProxy *HTTP01ChallengeProxySpec `json:"http01ChallengeProxy,omitempty"`
+}
+
+type HTTP01ChallengeProxySpec struct {
+    // enabled controls whether the HTTP01 challenge proxy is active.
+    // When enabled, the proxy redirects HTTP traffic from API:80 to ingress routers.
+    // +optional
+    Enabled bool `json:"enabled"`
+    
+    // internalPort specifies the internal port used by the proxy service.
+    // +kubebuilder:validation:Minimum=1024
+    // +kubebuilder:validation:Maximum=65535
+    // +optional
+    InternalPort int32 `json:"internalPort,omitempty"`  // default: 8888
+}
+```
+
+**Example Configuration:**
+
+```yaml
+apiVersion: config.openshift.io/v1
+kind: APIServer
+metadata:
+  name: cluster
+spec:
+  # existing fields...
+  http01ChallengeProxy:
+    enabled: true
+    internalPort: 8888
+status:
+  # existing status fields...
+  conditions:
+    - type: HTTP01ChallengeProxyReady
+      status: "True"
+      lastTransitionTime: "2025-05-12T00:00:00Z"
+      reason: "Initialized"
+      message: "HTTP01ChallengeProxy is ready"
+```
+    message: "Only one HTTP01ChallengeProxy instance named 'cluster' is allowed per cluster"
+```
+
+This design ensures that:
+- Only one `HTTP01ChallengeProxy` instance can exist cluster-wide
+- The instance must be named `cluster` to enforce the singleton pattern
+- The ValidatingAdmissionPolicy prevents creation of additional instances
+- The cluster-kube-apiserver-operator manages this singleton instance
+
+### Implementation Details/Notes/Constraints
+
+- The **cluster-kube-apiserver-operator** will be responsible for deploying and managing the proxy as a DaemonSet on control plane nodes that may host the API VIP.
+- **Deployment Conditions**: The operator deploys the proxy DaemonSet only when:
+  - `APIServer.spec.http01ChallengeProxy.enabled = true` is set in the cluster's APIServer configuration
+  - The cluster is running on a baremetal or supported platform (not cloud platforms with integrated DNS01 support)
+  - The operator validates that nftables/netfilter subsystem is available on target nodes
+- **Component Integration**: The operator will integrate the proxy lifecycle with existing API server management, ensuring consistent behavior and upgrade/downgrade strategies.
+- **Singleton Enforcement**: Only one proxy configuration per cluster is supported, enforced by the singleton nature of the APIServer CRD.
+- When the proxy starts, it checks which version of OCP it's running on. If it's 4.17+ it will proceed to configure MCO restart the nftables service rather than reboot the node when the file /etc/sysconfig/nftables.conf is modified. After that it creates the MC that configures the file + service.
```

Comment:
If this proxy support is being managed by a cluster-operator, it will only ever exist on an OpenShift version that the cluster operator is associated with. (i.e if this feature is added in 4.20, it will only ever exist on 4.20+ clusters.)

I suspect the proxy itself doesn't actually need to be aware of the OCP version it is running on anymore.

Maybe CKASO can be responsible for this re-configuration of the MCO as well?